### PR TITLE
feat(filter): add stateless/stateful mode routing for Responses API

### LINKED ISSUE #94: OpenAI Responses API: stateless pass-through mode
Forward Responses API requests to VLM/llm-d inference endpoints with zero body translation. Detect Responses API format (Items array vs Messages array) and route accordingly. Support previous_response_id chaining. No API translation before inference to maintain fidelity.

## Requirements
- Detect Responses API vs Chat Completions format from request body
- Pass-through routing to VLM/llm-d without body mutation
- Support encrypted reasoning items forwarding
- Streaming support for Responses API format
- Integration tests with mock Responses API endpoint

### LINKED ISSUE #217: Passthrough integration test
A simple POST /v1/responses with store=false and a string input hits inference directly with sub-millisecond proxy overhead. The filter chain only activates when the request actually needs stateful processing.

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Enhances the `openai_responses_format` filter to compute a combined `mode` (`stateless` | `stateful`) from request body fields, enabling single-condition branch chain routing for the Responses API fast path
- Adds `has_tools` and `has_prompt_id` detection to the classifier (reads nested `prompt.prompt_id` per OpenAI Prompts guide), promotes `mode` to metadata, filter_results, and a configurable `x-praxis-responses-mode` header
- A `prompt` object in the request body is now recognized as a Responses API format indicator
- Adds `responses-routing.yaml` example config demonstrating branch chain routing with a single backend
- Adds unit tests and integration tests covering all stateful conditions, stateless edge cases, and body-not-mutated verification
- Fixes proposal doc terminology: `conversation_id` → `conversation` (per OpenAI/OpenResponses spec)

Closes praxis-proxy/ai#217
Closes praxis-proxy/ai#94

## Test plan

- [x] `cargo test -p praxis-proxy-filter -- openai` — all unit tests pass
- [x] `cargo test -p praxis-tests-integration --test suite -- responses_routing` — 6 integration tests pass
- [x] `make lint` — clean
- [x] `make build` — clean
- [x] `make doc` — clean

### CODE REVIEW

File: filter/src/builtins/http/ai/openai/responses/config.rs [hunk 1]
```diff
@@ -58,6 +58,10 @@ pub(crate) struct ResponsesFormatHeaders {
     /// Header name for the extracted stream flag.
     #[serde(default = "default_stream_header")]
     pub stream: Option<String>,
+
```
Comment (shaneutt) [inline filter/src/builtins/http/ai/openai/responses/config.rs]: **[Important]** Doc comment says `passthrough` but the code emits `stateless`. The proposal doc was updated from `passthrough` to `stateless` in this same PR, and the rest of the codebase (tests, example config, filter results) all use `stateless`. This doc comment should say `stateless` to match.

```rust
/// Header name for the computed mode (`stateless` or `stateful`).
```

File: filter/src/builtins/http/ai/openai/responses/config.rs [hunk 2]
```diff
@@ -58,6 +58,10 @@ pub(crate) struct ResponsesFormatHeaders {
...
+    /// Header name for the computed mode (`passthrough` or `stateful`).
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/config.rs]: https://github.com/praxis-proxy/praxis/pull/445/changes/7a43eb24c7e3f59f436be15087527ac5509fc732

File: tests/integration/tests/suite/responses_routing.rs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        ),
+        (
+            "background=true",
+            r#"{"model":"gpt-4.1","input":"Hello","store":false,"background":true}"#,
+        ),
+        (
+            "conversation present",
+            r#"{"model":"gpt-4.1","input":"Hello","store":false,"conversation":{"id":"conv_123"}}"#,
+        ),
+    ];
```
[the remark this answers is not in the document]
Comment (leseb) [reply tests/integration/tests/suite/responses_routing.rs:44]: https://github.com/praxis-proxy/praxis/pull/445/changes/7a43eb24c7e3f59f436be15087527ac5509fc732

File: filter/src/builtins/http/ai/openai/responses/tests.rs [hunk 4]
```diff
@@ -242,10 +242,7 @@ async fn promotes_headers_for_full_responses_request() {
     assert_eq!(headers.get("x-praxis-ai-format"), Some(&"responses"), "format header");
     assert_eq!(headers.get("x-praxis-ai-model"), Some(&"gpt-4.1"), "model header");
     assert_eq!(headers.get("x-praxis-ai-stream"), Some(&"true"), "stream header");
-    assert!(
-        !headers.contains_key("x-praxis-ai-background"),
-        "background should not have a default header"
-    );
+    assert_eq!(headers.get("x-praxis-responses-mode"), Some(&"stateful"), "mode header");
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/tests.rs:266]: https://github.com/praxis-proxy/praxis/pull/445/changes/7a43eb24c7e3f59f436be15087527ac5509fc732

File: tests/integration/tests/suite/responses_routing.rs [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+/// Stateful requests (mode=stateful) enter the branch chain and route
+/// to `stateful_port`. Passthrough requests (mode=stateless) or
+/// non-Responses requests fall through to `default_port`.
+fn mode_branch_yaml(proxy_port: u16, stateful_port: u16, default_port: u16) -> String {
+    format!(
+        r#"
+listeners:
+  - name: default
+    address: "127.0.0.1:{proxy_port}"
+    filter_chains: [main]
```
[the remark this answers is not in the document]
Comment (leseb) [reply tests/integration/tests/suite/responses_routing.rs:156]: https://github.com/praxis-proxy/praxis/pull/445/changes/7a43eb24c7e3f59f436be15087527ac5509fc732

File: filter/src/builtins/http/ai/openai/responses/tests.rs [hunk 6]
```diff
@@ -398,12 +427,156 @@ async fn null_headers_suppress_emission() {
     );
 }
 
+// -----------------------------------------------------------------------------
+// Mode Computation
+// -----------------------------------------------------------------------------
+
+#[tokio::test]
+async fn mode_passthrough_when_store_false_no_stateful_markers() {
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/tests.rs]: https://github.com/praxis-proxy/praxis/pull/445/changes/7a43eb24c7e3f59f436be15087527ac5509fc732

File: filter/src/builtins/http/ai/openai/responses/classifier/mod.rs [hunk 7]
```diff
@@ -89,6 +94,10 @@ pub(crate) fn classify_request_body(body: &[u8]) -> ClassifiedRequest {
         background: obj.get("background").and_then(serde_json::Value::as_bool),
         has_previous_response_id: obj.get("previous_response_id").is_some_and(|v| !v.is_null()),
         has_conversation: obj.get("conversation").is_some_and(|v| !v.is_null()),
+        has_tools: obj
+            .get("tools")
+            .is_some_and(|v| v.as_array().is_some_and(|a| !a.is_empty())),
+        has_prompt_id: obj.get("prompt_id").is_some_and(|v| !v.is_null()),
```
Comment (nerdalert) [inline filter/src/builtins/http/ai/openai/responses/classifier/mod.rs]: I _think_ the prompt ID is in the wrong place according to https://developers.openai.com/api/docs/guides/prompting. Responses API prompt-template requests use prompt.prompt_id, not top-level prompt_id (OpenAI docs). That means a real request like {"store":false,"prompt":{"prompt_id":"pmpt_123"}} can miss the stateful route or be rejected as unknown. Can we read prompt.prompt_id here and also treat a non-null top-level prompt object as a Responses request?

Suggested:

  ```rust
  has_prompt_id: obj
      .get("prompt")
      .and_then(serde_json::Value::as_object)
      .and_then(|prompt| prompt.get("prompt_id"))
      .is_some_and(|v| !v.is_null()),
```

  And in `classify_format`:

```rust
  if obj.contains_key("input") || obj.get("prompt").is_some_and(serde_json::Value::is_object) {
      AiRequestFormat::Responses
  }
```

The example in the guide shows:

```curl
curl -s -X POST "https://api.openai.com/v1/responses" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $OPENAI_API_KEY" \
-d '{
    "prompt": {
    "prompt_id": "pmpt_123",
    "variables": {
        "city": "San Francisco"
    }
    }
}'
```

Other than that, looks awesome, Ty!

File: filter/src/builtins/http/ai/openai/responses/mod.rs [hunk 8]
```diff
@@ -176,11 +179,39 @@ fn handle_invalid_format(format: AiRequestFormat, config: &ResponsesFormatConfig
...
+/// and `Some("stateless")` when it can be forwarded directly to a
+/// native Responses backend. Returns `None` for non-Responses formats.
+fn compute_mode(classified: &classifier::ClassifiedRequest) -> Option<&'static str> {
+    if classified.format != AiRequestFormat::Responses {
+        return None;
+    }
+    // OpenAI spec: store defaults to true when omitted
+    let stateful = classified.has_previous_response_id
+        || classified.has_tools
+        || classified.store.unwrap_or(true)
```
Comment (shaneutt) [inline filter/src/builtins/http/ai/openai/responses/mod.rs:213]: 👍

File: filter/src/builtins/http/ai/openai/responses/classifier/mod.rs (hunk 7 above)
Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/classifier/mod.rs]: wow good catch thanks

Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/classifier/mod.rs]: 218a6047041c5cc5ba258678dcbdddda850e8b13

Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/classifier/mod.rs]: this is very strange, this page https://developers.openai.com/api/docs/guides/prompting says `prompt_id` but the official spec https://raw.githubusercontent.com/openai/openai-openapi/refs/heads/master/openapi.yaml only presents:

```
{
  "prompt": {
    "id": "pmpt_abc123",
    "version": "2",
    "variables": {}
  }
}
```

ohhhh:

```
curl-s -X POST ""https://api.openai.com/v1/responses"\
-H "Content-Type: application/json" \
-H "Authorization: Bearer $OPENAI_API_KEY" \
-d '{
    "prompt": {
    "prompt_id": "pmpt_123",
    "variables": {
        "city": "San Francisco"
    }
    }
}'

{
  "error": {
    "message": "Missing required parameter: 'prompt.id'.",
    "type": "invalid_request_error",
    "param": "prompt.id",
    "code": "missing_required_parameter"
  }
}%
```

the doc is invalid.

**Review by shaneutt — DISMISSED** [review]

File: filter/src/builtins/http/ai/openai/responses/classifier/mod.rs (hunk 7 above)
Comment (leseb) [reply filter/src/builtins/http/ai/openai/responses/classifier/mod.rs]: the commit never made it in main, i guess the pr was merged and the push happened after? anyways https://github.com/praxis-proxy/praxis/pull/694 fixed it :)
